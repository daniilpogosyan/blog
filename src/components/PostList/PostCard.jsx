import { Link } from 'react-router-dom';

import style from './PostCard.module.css';

export default function PostCard({ title, author, url, status }) {
  const statusTagClassName = `${style['status-tag']} ${style[status]}`;
  return (
    <div className={style.card}>
      <h2 className={style.heading}>{title}</h2>
      <p className={style.author}>by {author.username}</p>
      <Link to={url} className={style.link}>Read post</Link>
      <div className={statusTagClassName}>
        {status}
      </div>
    </div>
  )
}
