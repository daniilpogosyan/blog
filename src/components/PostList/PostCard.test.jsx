import { render } from "@testing-library/react";
import PostCard from "./PostCard";
import { MemoryRouter } from "react-router-dom";


test('PostCard structure', () => {
  const post = {
    title: "Post for test",
    author: {username: 'John Doe'},
    id: 'mockId12345',
    status: 'published',
    createdAt: new Date(1.5e12)
  }
  
  const postCard = render(
    <MemoryRouter>
      <PostCard {...post} />
    </MemoryRouter>
  );

  expect(postCard.container).toMatchSnapshot();
});
