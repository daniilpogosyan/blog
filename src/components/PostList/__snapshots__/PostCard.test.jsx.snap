// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard structure 1`] = `
<div>
  <div
    class="card"
  >
    <h2
      class="heading"
    >
      Post for test
    </h2>
    <p
      class="author"
    >
      by 
      John Doe
    </p>
    <a
      class="link"
      href="/posts/mockId12345"
    >
      Read post
    </a>
    <div
      class="status-tag published"
    >
      published
    </div>
    <span
      class="date"
    >
      Fri Jul 14 2017
    </span>
  </div>
</div>
`;
